homeassistant/util/secrets.py

```diff
@@ -0,0 +1,133 @@
+"""Module to help keep passwords and other secrets."""
+
+import os
+import logging
+import collections
+from enum import Enum
+from homeassistant.config import load_yaml_config_file
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+try:
+    import keyring  # pylint: disable=wrong-import-order,wrong-import-position
+except ImportError:
+    keyring = None
+
+# Contains secrets resolved by get_secret (from config files & components)
+# History can be used to edit secrets through the frontend after HA started.
```

Comment:
Ok, I started looking into secret setting through a cmdline switch, but this is probably not ideal?